File: UCSD/Data-Structures-and-Algorithms-Specialization/Course1-AlgorithmicToolbox/Assignment2/7-GreatestCommonDivisor/GCD.cpp
#include <iostream>
using namespace std;



long long gcd(long long a, long long b)
{
    if (a < b) return gcd(b, a);
    if (b == 0) return a;
    return gcd(b, a % b);
}



int main()
{
    long long a, b;
    cin >> a >> b;
    cout << gcd(a, b) << endl;
    return 0;
}

File: UCSD/Data-Structures-and-Algorithms-Specialization/Course1-AlgorithmicToolbox/Assignment2/7-GreatestCommonDivisor/GCD.java
import java.util.Scanner;



class GCD 
{
    static long gcd(long a, long b)
    {
        if (a < b) return gcd(b, a);
        if (b == 0) return a;
        return gcd(b, a % b);
    }



    public static void main(String[] args)
    {
        Scanner scanner = new Scanner(System.in);

        long a = scanner.nextLong();
        long b = scanner.nextLong();

        System.out.println(gcd(a, b));
    }
}
